### PR TITLE
[React utils refactor] delete react-utils

### PR DESCRIPTION
**Pull request by benoitgrelard**
>Note: This PR also removes the handling of `ownerDocument` and `ownerWindow` as it was only done in one place. Our intention is to address this globally once the requirement arises.

### CODE REVIEW

File: packages/react/alert-dialog/src/AlertDialog.tsx [hunk 1]
```diff
@@ -89,33 +68,39 @@ const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
...
-        aria-label={ariaLabel || undefined}
-        {...dialogContentProps}
-        ref={ref}
-        onOpenAutoFocus={composeEventHandlers(dialogContentProps.onOpenAutoFocus, (event) => {
-          event.preventDefault();
-          cancelRef.current?.focus({ preventScroll: true });
-        })}
-      />
-    </AlertDialogContentProvider>
+    <DialogContent
```
Comment (benoitgrelard) [inline packages/react/alert-dialog/src/AlertDialog.tsx:78]: best viewed with whitespace turned off

File: packages/react/scroll-area/package.json [hunk 2]
```diff
@@ -23,11 +23,11 @@
...
     "@radix-ui/react-primitive": "workspace:*",
     "@radix-ui/react-use-callback-ref": "workspace:*",
-    "@radix-ui/react-use-layout-effect": "workspace:*",
-    "@radix-ui/react-utils": "workspace:*"
+    "@radix-ui/react-use-layout-effect": "workspace:*"
   },
   "devDependencies": {
-    "@stitches/react": "canary"
+    "@stitches/react": "canary",
+    "@types/resize-observer-browser": "^0.1.4"
```
Comment (benoitgrelard) [inline packages/react/scroll-area/package.json:30]: I had missed that one

File: tsconfig.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
       "@radix-ui/rect": ["./packages/core/rect/src"],
       "@radix-ui/react-accessible-icon": ["./packages/react/accessible-icon/src"],
       "@radix-ui/react-accordion": ["./packages/react/accordion/src"],
+      "@radix-ui/react-alert-dialog": ["./packages/react/alert-dialog/src"],
```
Comment (benoitgrelard) [inline tsconfig.json:12]: not sure what happened to this one, but it seems it was either never there, or disappeared in a bad merge at some point

File: tsconfig.json [hunk 4]
```diff
@@ -36,7 +37,7 @@
       "@radix-ui/react-progress": ["packages/react/progress/src"],
       "@radix-ui/react-radio-group": ["./packages/react/radio-group/src"],
       "@radix-ui/react-roving-focus": ["./packages/react/roving-focus/src"],
-      "@radix-ui/react-scroll-area": ["./packages/react/react-scroll-area/src"],
+      "@radix-ui/react-scroll-area": ["./packages/react/scroll-area/src"],
```
Comment (benoitgrelard) [inline tsconfig.json:40]: same here, it was probably always wrong, discovered it when I included scroll area in the ssr tests

File: yarn.lock [hunk 5]
```diff
@@ -3648,17 +3648,6 @@ __metadata:
   languageName: unknown
   linkType: soft
 
-"@radix-ui/react-utils@workspace:*, @radix-ui/react-utils@workspace:packages/react/utils":
```
Comment (benoitgrelard) [inline yarn.lock:3651]: BYYYYYYYE!

File: packages/react/alert-dialog/src/AlertDialog.tsx [hunk 6]
```diff
@@ -89,33 +68,39 @@ const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
...
+      })}
+    >
+      <AlertDialogContentProvider cancelRef={cancelRef}>
+        {process.env.NODE_ENV === 'development' && <AccessibilityDevWarnings {...props} />}
+        {/**
+         * We have to use `Slottable` here as we cannot wrap the `AlertDialogContentProvider`
+         * around everything, otherwise the `AccessibilityDevWarnings` would be rendered straight away.
+         * This is because we want the accessibility checks to run only once the content is actually
+         * open and that behaviour is already encapsulated in `DialogContent`.
+         */}
```
Comment (jjenzz) [inline packages/react/alert-dialog/src/AlertDialog.tsx:100]: :+1:

File: tsconfig.json (hunk 3 above)
Comment (jjenzz) [reply tsconfig.json:12]: strange !

File: tsconfig.json (hunk 4 above)
Comment (jjenzz) [reply tsconfig.json:40]: These problems are making me wonder more and more why we need these if they weren't highlighted by anything

**Review by jjenzz — APPROVED** [review]

**Pull request merged**